presto-main/src/test/java/io/prestosql/metadata/TestInformationSchemaMetadata.java

```diff
@@ -210,12 +212,8 @@ public void testInformationSchemaPredicatePushdownWithConstraintPredicateOnSchem
         ConnectorMetadata metadata = new InformationSchemaMetadata("test_catalog", this.metadata);
         InformationSchemaTableHandle tableHandle = (InformationSchemaTableHandle)
                 metadata.getTableHandle(session, new SchemaTableName("information_schema", "schemata"));
-        tableHandle = metadata.applyFilter(session, tableHandle, constraint)
-                .map(ConstraintApplicationResult::getHandle)
-                .map(InformationSchemaTableHandle.class::cast)
-                .orElseThrow(AssertionError::new);
-
-        assertEquals(tableHandle.getPrefixes(), ImmutableSet.of(new QualifiedTablePrefix("test_catalog")));
+        Optional<ConstraintApplicationResult<ConnectorTableHandle>> result = metadata.applyFilter(session, tableHandle, constraint);
+        assertFalse(result.isPresent());
```

Comment:
please another test with a table where it actually works

Reply:
So the original test is aimed at a `schemata` table to make sure
> predicate on non tables enumerating table should not cause schemas to be enumerated.

After the change, a predicate push down can't be applied to "non tables enumerating tables", i.e., catalog-wise tables.

So the modified test actually preserved the logic of testing on catalog-wise tables. It looks like it's not really necessary to add another test which predicate pushdown actually works. But I did modify the code comments to reduce confusion.